### DISCUSSION TITLE
interpreter functionality

### DISCUSSION THREAD

**Discussion by goto40** [textX/textX#352] [Ideas]
`textx --help` reveals...
```
Commands:
  check            Check/validate model given its file path.
  generate         Run code generator on a provided model(s).
  list-generators  List all registered generators
  list-languages   List all registered languages
```

Thus, we can 
 * "**check**" (validate) a model
 * "**generate**" some code from a model

How about adding the ability to
 * "**run**" a model (run some interpret like functionality. Alternatively we could also call it "**interpret**".

With this we could then `textx run --grammar Flow.tx --target simulation mymodel.flow` or just `textx run --target simulation mymodel.flow`.

Any ideas? What do you think?

**Comment by igordejanovic** [comment]
I think it does make sense as we already have `generate` and `list-generators`. Interpreters are just another way to specify language semantics, so it would make sense to have `interpret` and `list-interpreters`.

The only consideration is that while `generate` is normally part of the build process, `interpret` is not, it is a part of the deployment process. So we have to discuss whether it is beneficial to have a standard CLI way of running interpreters in the production.

**Comment by goto40** [comment]
I completely agree with you! I even think that it is "better" for an end user to have a new (independent of textX CLI) application. Like you have "python script.py", you could have "mylang script.mylang"...

The only point which might be interesting is to be able to provide a "run" button in an IDE...

Let this idea mature a little more... (it has no high prio in my eyes).
